### DISCUSSION TITLE
Problem with Xray(VLESS+Vision+TLS)+Caddy example

### DISCUSSION THREAD

**Discussion by houmie** [lxhao61/integrated-examples#162] [Q&A] [answered]
I have tried for many hours to get this [example](https://github.com/lxhao61/integrated-examples/tree/main/Xray(VLESS%2BVision%2BTLS)%2BCaddy) working. 

Major problem is the storage path:

```
"storage": {
    "module": "file_system",
    "root": "/root/tls"
  }
```

I'm logged in as root on Debian.  My home drive is "/root/"

Caddy has no permission to access it. So I have to allow it.
```
mkdir ~/tls
chown caddy:caddy ~/tls
```

But still starting caddy, doesn't generate the certificates in that folder.

**Caddy shows this error:**
```
Jun 08 17:28:24 gymx caddy[16147]: Error: loading initial config: loading new config: tls app module: start: automate: managing [gymx.com]: automate: manage [gymx.com]: gymx.com: caching certificate: open /root/tls/certificates/acme-v02.api.letsencrypt.org-directory/gymx.com/gymx.com.key: permission denied
```

Maybe I misunderstood it. Do I have to create a user called ```tls``` with home /home/tls ?
Then change the user for caddy and xray to `tls`?

Thanks

**Comment by lxhao61** [comment, marked as answer]
试试命令：
```
chown -R caddy:caddy ~/tls
```
另外Caddy存储配置为/root/tls，用户及组配置为caddy根本就是多余，直接配置为root不是更好吗？
